fixed attractions page bug not keeping different park days separate, …

achilders314 opened the pull request:
…added footer test, TODO fix Attractions test

achilders314 committed:
fixed attractions page bug not keeping different park days separate, added footer test, TODO fix Attractions test

OleDakotaJoe commented:
When make pull requests, its always a good idea to get detailed to answer a few key questions.
1. Summary
2. Why was it necessary?
3. What did you actually do?
4. How did you test it?
5. Include screenshots.

OleDakotaJoe commented on .firebase/hosting.YnVpbGQ.cache:
```diff
@@ -2,21 +2,21 @@
 castle-favicon.png,1672178909753,4c9e9d9d55e59ac44f3473bc1ba10209faaf3120fde07625b35e49ded7984f76
 manifest.json,1671416298659,27704a75f8003f4102ff24935dedc4e12c83ec0b61b88de3bb08ef08def39672
 robots.txt,1670778116306,bfe106a3fb878dc83461c86818bf74fc1bdc7f28538ba613cd3e775516ce8b49
-asset-manifest.json,1673735773941,1db97b8231b7c2760c8f3210a008f0cc997bb328e994a8b80becfef124747bf3
```
I'm not actually familiar with this file, I'm curious - was it intended to be committed?

OleDakotaJoe replied:
Actually, just looked -- looks like the firebase cache should actually be added to your .gitignore file.

Check it out: https://stackoverflow.com/questions/52130772/whats-the-purpose-of-firebase-hosting-alphanum-cache

achilders314 replied:
I'm not actually sure how this one changed ... I'll take a look though! Thanks!

OleDakotaJoe commented on functions/index.js:
```diff
@@ -5,26 +5,26 @@ const request = require("request");
 admin.initializeApp(functions.config().firebase);
 
-// const usersRef = admin.firestore().collection("users");
+// exports.helloWorld = functions.https.onRequest((request, response) => {
```
Is there a reason why this commented-out code is left in the codebase? 

I find it to be good "code hygiene" to go ahead and remove commented out code, especially because if you consistently use PRs you will ALWAYS have that code in the git blame and history.

achilders314 replied:
Got it! Thanks! I just remembered to go back and look at that function, it was a default one when I set up the firebase project at the beginning.

OleDakotaJoe commented on functions/index.js:
```diff
@@ -52,9 +51,10 @@ exports.collectParkData = functions.pubsub
-            const currentWaitTime = attraction.queue && attraction.queue["STANDBY"] && attraction.queue["STANDBY"].waitTime ? attraction.queue["STANDBY"].waitTime : 0;
+            const currentWaitTime = attraction.queue && attraction.queue["STANDBY"] && attraction.queue["STANDBY"].waitTime ?
+                                    attraction.queue["STANDBY"].waitTime : 0;
             // eslint-disable-next-line max-len
```
I know this is not in the PR, but I'm curious why we are disabling, and if there is another way we could handle the linter?

achilders314 replied:
There were a few lines that ended up with too many characters despite me breaking them up the best I could. There was an 80 character max I think per line?

OleDakotaJoe replied:
You can return right before each operator and it's pretty standard code style

```
const currentWaitTime = attraction.queue 
&& attraction.queue["STANDBY"] 
&& attraction.queue["STANDBY"].waitTime 
     ? attraction.queue["STANDBY"].waitTime 
     : 0;

```

OleDakotaJoe replied:
You can also use something called "optional chaining", I'm assuming that you're using the multiple && operators to ensure you don't get a null pointer error. 



```
const currentWaitTime = attraction?.queue?["STANDBY"]?.waitTime || 0 ;
```

OleDakotaJoe replied:
Btw you'd have to verify how I did the optional chaining on the array access, I'm not 100% sure on the syntax off the top of my head

achilders314 replied:
Yes I think I've tried optional chaining in this app a couple different times but it wasn't working for some reason. I'll have to give it another go, I know it would be a lot cleaner.

OleDakotaJoe commented on src/components/Attractions.test.js:
```diff
@@ -0,0 +1,80 @@
+let noUserData = {};
+
+describe.skip('Attractions Basic Setup', () => {
+    // const { loading } = React.useContext(useAuth);
```
Potential cleanup item?

achilders314 replied:
I'm planning on possibly using these when I get the testing working.

OleDakotaJoe commented on src/components/Attractions.test.js:
```diff
@@ -0,0 +1,80 @@
+
+    ];
+
+    // Still can't get this one to render correctly
```
If you need help getting the testing working, lets definitely get on a call and we can pair program through it!

achilders314 replied:
YESSSSSS please :)

OleDakotaJoe replied:
Sure thing, when are you available?

achilders314 replied:
This week I can do M/W/Th/F any time in the 1:30-4:30 range. If it needs to be evening, I can do usually Wednesday or Friday evenings? Or weekend days usually from about 1-3ish?

Any of that strike your fancy?

OleDakotaJoe replied:
Let's do Monday

OleDakotaJoe commented on src/components/Attractions.test.js:
```diff
@@ -0,0 +1,80 @@
+    )
+    
+
+    // test.map((userTest) => {
```
potential cleanup item, or maybe we can implement this?

achilders314 replied:
I've tried lots of things but haven't been successful so far.

achilders314 commented:
Hi Steven! Thanks for this, I didn't really know the normal protocol for a PR.

Summary
-I added my first unit test of the Footer. I started adding more to other components but am having trouble because they use context.
-I fixed a bug where if I selected 4 attractions on the first day of my trip, any other day with that same park would exclude those same attractions. It's resolved now.

What did you actually do?
I updated the code to call the loadUserAttractions function on line 43 to fix the bug. The day was switching but the attractions weren't reloading.

How did you test it?
I verified all of the attraction days were displaying differently with different attractions missing, and the My Trip page also matched up with this.

Include screenshots.
![Screenshot_20230127_050410](https://user-images.githubusercontent.com/102301042/215222549-a8765c98-c176-40d2-92d7-3395c3fc2700.png)
![Screenshot_20230127_050446](https://user-images.githubusercontent.com/102301042/215222552-1af7e8cb-81cc-4f92-8d6e-39437075d018.png)
![Screenshot_20230127_050502](https://user-images.githubusercontent.com/102301042/215222554-8f304360-81de-442a-b46a-353eb51e4dda.png)
![Screenshot_20230127_050517](https://user-images.githubusercontent.com/102301042/215222555-b3a58ae0-8dd9-4c14-b7a4-0717430b136a.png)

OleDakotaJoe commented on src/components/Footer.test.js:
```diff
@@ -0,0 +1,14 @@
+    it('Displays my copyright message', () => {
+        render(<Footer/>);
+
+        expect(screen.getByText("2023 by Alison Childers", {
```
Would reccomend mocking, or using something called dependency injection/inversion of control to pass either the date INTO the component, or simply mock the Date function that you've used inside the component. Can definitely explain these concepts with you on a call if you like! 


Here is a link so you can read up on your own:
 https://softwareengineering.stackexchange.com/questions/140992/is-dependency-injection-essential-for-unit-testing

achilders314 replied:
Ah ok yes this makes sense! Thanks!

OleDakotaJoe reviewed (commented):
Loving the introduction of testing here!

Would love to see the testing in Attractions.test.js be functional, and you have a few items to cleanup.
